File: src/main/java/frc/robot/commands/intake/Intake.java
package frc.robot.commands.intake;

import edu.wpi.first.wpilibj2.command.CommandBase;
import frc.robot.OI;
import frc.robot.subsystems.IntakeSubsystem;
import frc.robot.subsystems.IntakeSubsystem.Side;

public class Intake extends CommandBase {
    private IntakeSubsystem intake;

    /**
     * Creates a new ExampleCommand.
     *
     * @param subsystem The subsystem used by this command.
     */
    public Intake(IntakeSubsystem intake) {
        this.intake = intake;

        addRequirements(intake);
    }

    @Override
    public void initialize() {
        
    }

    @Override
    public void execute() {
        // TODO: Doing controller logic/bindings in commands is the eighth deadly sin, so this needs to change
        if (OI.operatorController.getLeftBumper()) {
            intake.extendIntake(Side.leftIntake);
        } else {
            intake.retractIntake(Side.leftIntake);
        }

        if (OI.operatorController.getRightBumper()) {
            intake.extendIntake(Side.rightIntake);
        } else {
            intake.retractIntake(Side.rightIntake);
        }
    }

    @Override
    public void end(boolean interrupted) {
        intake.retractIntake();
    }

    @Override
    public boolean isFinished() {
        return false;
    }
}

File: src/main/java/frc/robot/commands/intake/Intake.kt
package frc.robot.commands.intake

import frc.robot.subsystems.IntakeSubsystem
import edu.wpi.first.wpilibj2.command.CommandBase
import frc.robot.OI

class Intake(private val intake: IntakeSubsystem) : CommandBase() {
    init {
        addRequirements(intake)
    }

    override fun execute() {
        // TODO: Doing controller logic/bindings in commands is the eighth deadly sin, so this needs to change
        if (OI.operatorController.leftBumper) {
            intake.extendIntake(IntakeSubsystem.Side.leftIntake)
        } else {
            intake.retractIntake(IntakeSubsystem.Side.leftIntake)
        }

        if (OI.operatorController.rightBumper) {
            intake.extendIntake(IntakeSubsystem.Side.rightIntake)
        } else {
            intake.retractIntake(IntakeSubsystem.Side.rightIntake)
        }
    }

    override fun end(interrupted: Boolean) {
        intake.retractIntake()
    }

    override fun isFinished(): Boolean {
        return false
    }
}